### ISSUE TITLE
imagetest_harness_docker - registries.auth issue

### ISSUE THREAD

**Issue by rzsita** [chainguard-dev/terraform-provider-imagetest#107]
Hi,
I'm trying to get my private container repository to work by adding below.

Terraform code images/apko/tests/main.tf:
```hcl
terraform {
  required_providers {
    imagetest = { source = "chainguard-dev/imagetest" }
  }
}

variable "digest" {
  description = "The image digest to run tests over."
}

variable "imagetest_registries" {
  type        = map(object({
    auth = object({
      username = string
      password = string
    })
  }))
  description = "The registries auth when testing the image."
}

data "imagetest_inventory" "inventory" {}

resource "imagetest_harness_docker" "docker" {
  name      = "docker"
  inventory = data.imagetest_inventory.inventory
  registries = var.imagetest_registries

  envs = {
    IMAGE_NAME : var.digest
  }
}

resource "imagetest_feature" "test" {
  name    = "test"
  harness = imagetest_harness_docker.docker

  steps = [{
    name = "basic test"
    cmd  = "docker run --rm $IMAGE_NAME version"
  }]
}
``` 

Error:
```shell
╷
│ Error: Value Conversion Error
│ 
│   with module.apko.module.test-latest.imagetest_harness_docker.docker,
│   on images/apko/tests/main.tf line 26, in resource "imagetest_harness_docker" "docker":
│   26:   registries = var.imagetest_registries
│ 
│ An unexpected error was encountered trying to convert tftypes.Value into
│ provider.DockerRegistryResourceModel. This is always an error in the
│ provider. Please report the following to the provider developer:
│ 
│ mismatch between struct and object: Object defines fields not found in
│ struct: tls.
╵
``` 

Please advise.

Thanks so much in advance.

**Comment by joshrwolf** [comment]
hi @rzsita, thanks for opening an issue. I just want to strongly emphasize that we don't recommend using this provider for anything except the [chainguard images repo](https://github.com/chainguard-images/images). it is custom built specifically for that, and makes a lot of assumptions that don't translate into your typical terraform use case.

that being said, you've identified a bug with an existing knob we've exposed (but don't currently use), so thank you! we'll have a PR out shortly with the fix.

**Comment by stormqueen1990** [comment]
This is fixed via #108
